### PR TITLE
Extract services

### PR DESCRIPTION
**Pull request by mpndev**

### CODE REVIEW

File: web/modules/custom/custom_services/custom_services.info.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name: Custom Services
```
Comment (pafa7a) [inline web/modules/custom/custom_services/custom_services.info.yml]: That is not the Drupal way to define services. We should avoid creating modules that only contains services. These services should be defined in their respective module/s

File: web/modules/custom/custom_services/src/Event.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Drupal\custom_services;
+
+
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/Event.php]: Remove this extra line

File: web/modules/custom/custom_services/src/Event.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use Drupal\node\Entity\Node;
+use Drupal\paragraphs\Entity\Paragraph;
+
+class Event {
+
+  public $event = NULL;
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/Event.php]: Missing comments above the variables

File: web/modules/custom/custom_services/src/Event.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  public function id() {
+    return $this->event->id();
+  }
+
+  /**
+   * @param $field
+   *
+   * @return mixed
+   */
+  public function __get($field) {
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/Event.php]: This function is never used

File: web/modules/custom/custom_services/src/ShoppingCart.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\custom_services;
+
+use Drupal\custom_services\Event;
+class ShoppingCart {
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/ShoppingCart.php]: Missing one empty line between the last 'use' and the class definition

File: web/modules/custom/custom_services/src/ShoppingCart.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  private $shopping_cart = NULL;
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/ShoppingCart.php]: missing comments above the parameters, you should keep 1 empty line between them

File: web/modules/custom/custom_services/src/ShoppingCart.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    $tickets = $this->getTickets();
+
+    foreach ($tickets as &$ticket) {
+      if ($this->ticketIsFromThisEvent($ticket, $event)) {
+        $ticket['tickets_quantity']++;
+        $this->state_variables['have_tickets_for_this_event'] = TRUE;
+        break;
+      }
+    }
+    if ($this->state_variables['have_tickets_for_this_event'] === FALSE) {
```
Comment (pafa7a) [inline web/modules/custom/custom_services/src/ShoppingCart.php]: this can be 
`if (!$this->state_variables['have_tickets_for_this_event']) {`

**Review by pafa7a — CHANGES_REQUESTED** [review]

File: web/modules/custom/custom_services/src/Event.php (hunk 4 above)
Comment (mpndev) [reply web/modules/custom/custom_services/src/Event.php]: https://github.com/mpndev/shano/blob/extract_services/web/modules/custom/shano_shopping_cart/src/Form/AddTicketToCartForm.php   on line 98. We will use on other pleases too.

File: web/modules/custom/shano_shopping_cart/src/CustomClasses/ShoppingCart.php [hunk 8]
```diff
@@ -38,25 +44,25 @@ public function addTicketForEvent(Event $event) {
...
       }
     }
-    if ($this->state_variables['have_tickets_for_this_event'] === FALSE) {
+    if (!$this->state_variables['have_tickets_for_this_event']) {
       $tickets[] = [
-        'event_id' => $this->event->id(),
+        'event_id' => $event->id(),
         'tickets_quantity' => 1,
       ];
+      $tickets;
```
Comment (pafa7a) [inline web/modules/custom/shano_shopping_cart/src/CustomClasses/ShoppingCart.php]: please remove that. Most probably its a leftover from debugging

**Review by pafa7a — CHANGES_REQUESTED** [review]

**Review by pafa7a — APPROVED** [review]

**Pull request merged**